crates/dojo-core/src/execution_context.cairo

```diff
@@ -0,0 +1,11 @@
+use dojo_core::interfaces::IWorldDispatcher;
+use dojo_core::auth::components::AuthRole;
+use starknet::{ContractAddress, ClassHash};
+
+#[derive(Copy, Drop, Serde)]
+struct Context {
+    world: IWorldDispatcher, // Dispatcher to the world contract
+    caller_account: ContractAddress, // Address of the origin
+    caller_system: ClassHash, // Class hash of the calling system
```

Comment:
I think we probably want a system name here as well, since the class can be changed

```suggestion
    caller_system: ShortString, // Name of the calling system
    caller_system_class_hash: ClassHash, // Class hash of the calling system
```